system/Test/CIUnitTestCase.php

```diff
@@ -365,6 +363,21 @@ public function assertLogged(string $level, $expectedMessage = null)
         return $result;
     }
 
+    /**
+     * Asserts that there is a log record that contains `$logMessage` in the message.
+     */
+    public function assertLogContains(string $level, string $logMessage, string $message = ''): void
```

Comment:
My understanding is `$expectedMessage` is used since the assertion compares the message as expected in the logs.
For `assertLogContains`, I think it is counter-intuitive to use `$expectedMessage` when it can only be a part of the whole expected message. I was also thinking whether to use `$needle` or `$needleMessage` as alternative name.

Reply:
No preference from me

Reply:
I am a little uncomfortable with both $needle and $needleMessage. 
needle needs haystack.

So let's leave it as it is.